### DISCUSSION TITLE
scaling factors rather than mm for xy compensation?

### DISCUSSION THREAD

**Discussion by blex-max** [supermerill/SuperSlicer#2083] [General]
Would it be possible to allow for compensating XY (outer, inner, hole) via a percentage rather than specified mm? There's a few threads on the reprap forums for example about the exact scaling factors for getting holes to print at the correct size, as a linear adjustment doesn't tend to work so well. This is particularly true for small holes. I assume it might be possible since shrinkage seems to do about the same thing?

**Comment by supermerill** [comment]
A % of what? nozzle diameter? It doesn't make sense.
If you want to scale the model, scale it with the scale tool.

**Reply by blex-max** [reply]
I agree, it would make no sense to scale holes over nozzle diameter! I guess I was unclear, I am asking if it would be possible to implement something that scaled the size of holes in the XY plane **by a percentage of their diameter**. See this [thread](https://reprap.org/forum/read.php?262,407139,page=1) at reprap forum's (there are many others also), and particularly this post with regard to why a scaling factor/percentage system for hole size would be useful:

> The problem has been acknowledged for some time now, and I have two formulae that I acquired from a professional 3d printing firm. I can't remember who they were now, as I didn't make a note at the time.

> They suggested that where X mm was the required diameter of the hole then Y mm was the dimension needed to achieve it and that:

> a) for vertical holes Y = 1.0155X + 0.2795 mm

> b) for horizontal holes Y = 0.9927X + 0.3602 mm

> These figures were for PLA, and as I understand it, they take into account various factors including the fact that the inside of the hole is unrestrained against shrinkage.


Hole sizing such a long term, known problem in 3D printing, I think it could be a real jump if superslicer was the one to fix it! I might even try to learn C++ finally to help attempt it

**Reply by supermerill** [reply]
Can't find the thread, but I have already somebody who wanted this.
I asked him to provide me with data to support his claim.
So he printed X holes with varying diameters.
The difference in diameter couldn't be linked to the diameter. In fact, it was almost random up to a certain diameter where it becomes fixed. So currently the best approximation, from our tests, is a fixed value.

If you can provide me with a good protocol and tests values, I'm inclined to replicate & implement the new feature if conclusive.

There is a polyhole function if you really need to have an inner diameter that can't go lower than the desired value.

**Reply by blex-max** [reply]
Ok, I will investigate - thanks
